CompilerSource/compiler/jdi_utility.cpp

```diff
@@ -51,40 +51,41 @@ static void visit_overload(
  */
 static void iterate_overloads(
     definition_function* d, unsigned &min, unsigned &max, definition *varargs_t) {
-  for (map<arg_key, definition_function*>::iterator iter = d->overloads.begin();
-       iter != ((definition_function*)d)->overloads.end(); iter++) {
+  for (map<arg_key, definition_overload*>::iterator iter = d->overloads.begin();
+       iter != d->overloads.end(); iter++) {
     visit_overload(iter->second, min, max, varargs_t);
```

Comment:
You have ```    overload_map overloads; ///< Standard overloads, checked before template overloads.
    vector<definition_template*> template_overloads; ///< Array of reference stacks for each overload of this function.``` in class definition_function but you only seem to iterate the regular overloads. Even attempting to iterate the template_overloads vector I see 1 element but ```definition_template*``` has 0 specializations and 0 instantiations therefore,  I cannot get an accurate arg count. I think this is root cause of sqr expects 0 args but provided 1 issue but I'm at a loss on how to fix it.